.github/workflows/kuksa_databroker-cli_build.yml

```diff
@@ -53,8 +53,19 @@ jobs:
 
     steps:
     - uses: actions/checkout@v3
-      with: 
-        submodules: recursive
+
+    # We need tags for versioning,
+    # Checkout action above would only get them 
+    # using fetch-depth=0 (fetch all)
+    # just fetching tags alone is not sufficient as we need to
+    # have their history for them to be visible by git describe
+    # As a compromise we only deepen the shallow checkout.
+    # See also  https://github.com/actions/checkout/issues/701
+    - name: Making GIT tags available
```

Comment:
Since we are going to use the same version for all applications in the repo, I think it would make more sense to make the version globally available as an environment variable. That could be a simple github action running the equivalent of `git describe --tags --dirty=-dirty`.

Otherwise, we have to (re-)implement these extractions for C++ / python / Rust, and I'm not sure there's much benefit to doing that.